pkg/services/secrets/manager/manager.go

```diff
@@ -203,7 +199,7 @@ func (s *SecretsService) EncryptWithDBSession(ctx context.Context, payload []byt
 // currentDataKey looks up for current data key in cache or database by name, and decrypts it.
 // If there's no current data key in cache nor in database it generates a new random data key,
 // and stores it into both the in-memory cache and database (encrypted by the encryption provider).
-func (s *SecretsService) currentDataKey(ctx context.Context, label string, scope string, sess *xorm.Session) (string, []byte, error) {
+func (s *SecretsService) currentDataKey(ctx context.Context, label string, scope string) (string, []byte, error) {
 	// We want only one request fetching current data key at time to
 	// avoid the creation of multiple ones in case there's no one existing.
 	s.mtx.Lock()
```

Comment:
I think the use of this mutex should be replaced by the distributed lock used on `kvstore`, to guarantee a more predictable secrets manager, as discussed with @leandro-deveikis 

However, as it's not critical in order to fix the aforementioned issue, I'd prefer to leave that change for a following up PR.

Reply:
I think we need to talk and test it. for an HA environment, this in memory lock would not work of course, but adding a lock in the database would mean a few more operations on the database, which would impact the performance directly